internal/replicate/replicate.go

```diff
@@ -0,0 +1,8 @@
+package replicate
+
+import "context"
+
+type Replicator interface {
+	// Replicate copies images from the source registry to the local registry.
+	Replicate(ctx context.Context, ref string) error
```

Comment:
We may need to rethink the replicator in the future, if multiple implementations are going to be pulling the image. If this is the case we probably want to pass a reader to avoid duplicating the image pulling logic.